src/test/java/com/ericsson/ei/jmespath/test/TestJmesPathInterface.java

```diff
@@ -18,83 +18,138 @@
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputDiffpath), "UTF-8");
+            expectedResult = JsonNodeFactory.instance.booleanNode(false);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        String processRule = "incomplete_path_contains(@, 'testCase.StartedTime','3.56')";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void testIncompletePathContainsFunctionNotFirstMember() {
+        String jsonInput = null;
+        JsonNode expectedResult = null;
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputDiffpath), "UTF-8");
+            expectedResult = JsonNodeFactory.instance.booleanNode(true);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        String processRule = "incomplete_path_contains(@, 'testCaseStartedTime','3.56')";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void testIncompletePathContainsFunctionFirstMember() {
+        String jsonInput = null;
+        JsonNode expectedResult = null;
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputDiffpath), "UTF-8");
+            expectedResult = JsonNodeFactory.instance.booleanNode(true);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        String processRule = "incomplete_path_contains(@, 'testCaseExecutions','3.56')";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void testIncompletePathContainsFunctionSeveralMembers() {
+        String jsonInput = null;
+        JsonNode expectedResult = null;
+        try {
+            jsonInput = FileUtils.readFileToString(new File(inputDiffpath), "UTF-8");
+            expectedResult = JsonNodeFactory.instance.booleanNode(true);
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        String processRule = "incomplete_path_contains(@, 'fileInformation.extension','jar')";
+        JsonNode result = unitUnderTest.runRuleOnEvent(processRule, jsonInput);
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void testLiteral() {
+        JsonNode literalJson;
+        try {
+            literalJson = mapper.readTree("{}");
+            JsonNode input = mapper.readTree("{\"id\":\"test\"}");
+            ((ObjectNode) literalJson).put("eventId", "b6ef1hd-25fh-4dh7-b9vd-87688e65de47");
+            String ruleString = literalJson.toString();
+            ruleString = "`" + ruleString + "`";
+            JsonNode expectedResult = unitUnderTest.runRuleOnEvent(ruleString, input.toString());
+            assertEquals(expectedResult, literalJson);
```

Comment:
Will this test ever fail if the assert is within the try catch block?
Also the // TODO should be fixed.